lib/plugins/aws/package/lib/core-cloudformation-template.json

```diff
@@ -30,7 +30,10 @@
               "Principal": "*",
               "Resource": [
                 {
-                  "Fn::Join": ["", ["arn:aws:s3:::", { "Ref": "ServerlessDeploymentBucket" }, "/*"]]
+                  "Fn::Join": [
+                    "",
+                    ["arn:${AWS::Partition}:s3:::", { "Ref": "ServerlessDeploymentBucket" }, "/*"]
```

Comment:
I'm not 100% sure if this would conflict with our Serverless Variables when the object is created from this file (it shouldn't, correct?) and apparently we've used this pattern in other places.

However here's another (untested) way how this could be achieved with a `Ref`:

```javascript
'Fn::Join': [
  '',
  [
    'arn:',
    { Ref: 'AWS::Partition' },
    ':s3:::',
    { Ref: 'ServerlessDeploymentBucket' },
    '/*',
  ],
],
```

Reply:
> I'm not 100% sure if this would conflict with our Serverless Variables when the object is created from this file (it shouldn't, correct?) 

Variables are resolved before lifecycle engine is initialized, and this template is loaded within lifecycle flow, therefore to my understanding this doesn't affect in any way variables resolver.

I've also tested it with deployment, all works great

Reply:
Yes, that was what I was thinking as well. Just wanted to bring this to our attention. Good to hear that you've tested it and it works as expected :+1:

Reply:
It appears I didn't test it well after all (either I was on other branch, or tested just packaging).

Integration tests now crash badly: https://travis-ci.org/serverless/serverless/jobs/608836999

We should use what you've proposed, fix is ready here: https://github.com/serverless/serverless/pull/6937 (now definitely tested right way)